### PR TITLE
Create User and Profile  models or  APIs

### PR DESCRIPTION
**Pull request by harmeshverma01**

### CODE REVIEW

File: user/views.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from functools import partial
```
Comment (pythexcel) [inline user/views.py:1]: remove it

File: user/views.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    def get(self, request, id=None):
+        user = User.objects.all()
+        serializer = self.serializer_class(user, many=True)
+        return Response(serializer.data)
+    
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
```
Comment (pythexcel) [inline user/views.py:30]: This post method is wrong, why the token isn't created?

File: user/views.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+from rest_framework.views import APIView
+from .utils import admin_required
+from rest_framework import status
+from user.models import Profile, User
+
+# Create your views here.
+
+class Userview(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated, admin_required]
```
Comment (pythexcel) [inline user/views.py:16]: How the user will create an account if it is needed to authenticate first?

File: user/views.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  
+    def patch(self, request, id=None):
+        try:
+            user = User.objects.get()
+            serializer = self.serializer_class(user, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+            return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+        except:
+            return Response({"message": "User not Found"}, status=status.HTTP_404_NOT_FOUND)
```
Comment (pythexcel) [inline user/views.py:50]: This method will not work, Test it and fix it.

File: user/views.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            serializer = self.serializer_class(user, data=request.data, partial=True)
+            if serializer.is_valid():
+                serializer.save()
+            return Response(serializer.data, status=status.HTTP_202_ACCEPTED)
+        except:
+            return Response({"message": "User not Found"}, status=status.HTTP_404_NOT_FOUND)       
+
+
+class LoginView(APIView):
+    serializer_class = LoginSerializer
```
Comment (pythexcel) [inline user/views.py:54]: What's the need for this serializer here?

File: user/views.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+             token = Token.objects.get_or_create(user=user)
+             return Response({'token': str(token[0])})
+         return Response({'details':'User Not Found'}, status=status.HTTP_404_NOT_FOUND)
+
+
+
+class UserProfile(APIView):
+    serializer_class = ProfileSerializer
+    
+    def get(self, request, id=None):
```
Comment (pythexcel) [inline user/views.py:68]: This method is wrong.

File: user/views.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+         user = authenticate(email=request.data.get('email'), password=request.data.get('password'))
+         if user:
+             token = Token.objects.get_or_create(user=user)
+             return Response({'token': str(token[0])})
+         return Response({'details':'User Not Found'}, status=status.HTTP_404_NOT_FOUND)
+
+
+
+class UserProfile(APIView):
+    serializer_class = ProfileSerializer
```
Comment (pythexcel) [inline user/views.py:66]: No authentication check.

File: user/views.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        if serializer.is_valid():
+            serializer.save()
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+    
+    
+    def patch(self, request, id=None):
+        try:
+            user = Profile.objects.get(id=id)
+            serializer = self.serializer_class(user, data=request.data, partial=True)
+            return Response(serializer.data, status=status.HTTP_206_PARTIAL_CONTENT)
```
Comment (pythexcel) [inline user/views.py:85]: status code shouldn't be 206

File: user/views.py (hunk 1 above)
Comment (harmeshverma01) [reply user/views.py:1]: fixed

File: user/views.py (hunk 2 above)
Comment (harmeshverma01) [reply user/views.py:30]: fixed

File: user/views.py (hunk 3 above)
Comment (harmeshverma01) [reply user/views.py:16]: fixed

File: user/views.py (hunk 5 above)
Comment (harmeshverma01) [reply user/views.py:54]: fixed

File: user/views.py (hunk 4 above)
Comment (harmeshverma01) [reply user/views.py:50]: fixed

File: user/views.py (hunk 7 above)
Comment (harmeshverma01) [reply user/views.py:66]: fixed

File: user/views.py (hunk 8 above)
Comment (harmeshverma01) [reply user/views.py:85]: fixed

File: user/manager.py [hunk 9]
```diff
@@ -24,14 +23,16 @@ def staff_user(self, name, email, password):
...
-            password=password
+            password=password,
+            role='admin'
         )
         user.is_staff = True
-        user.admin = True
+        user.is_admin = True
+        user.is_superuser = True
         user.save(using=self._db)
+        print(user.__dict__)
```
Comment (TACGithub) [inline user/manager.py:36]: remove this

**Pull request merged**